internal/rpc/tesla_service.go

```diff
@@ -48,6 +61,60 @@ func (t *TeslaRPCService) RegisterNewSyntheticDevice(ctx context.Context, req *g
 	return &grpc.RegisterNewSyntheticDeviceResponse{}, nil
 }
 
+func isVINChar(c rune) bool {
+	// Of course, there is more subtlety to this: disallowed characters, among other things.
+	return 'A' <= c && c <= 'Z' || '0' <= c && c <= '9'
+}
+
+func (t *TeslaRPCService) RegisterNewSyntheticDeviceV2(ctx context.Context, req *grpc.RegisterNewSyntheticDeviceV2Request) (*grpc.RegisterNewSyntheticDeviceV2Response, error) {
+	if len(req.Vin) != 17 {
+		return nil, status.Error(codes.InvalidArgument, "VIN does not have length 17")
+	}
+	for i, c := range req.Vin {
+		if !isVINChar(c) {
+			return nil, status.Error(codes.InvalidArgument, fmt.Sprintf("VIN has invalid character %q at position %d", c, i))
+		}
+	}
```

Comment:
I don't know if we care about minting Tesla's in Japan

Reply:
I thought about this. I have yet to see them actually differ here